MinPlatform/TestPointCheckLib: Fix incorrect modification of MemoryTypeInformation HOB and memory leak

joschock opened the pull request:
DxeCheckMemoryTypeInformation - 
fix an issue where MemoryTYpeInfo check was incorectly modifying the MemoryTypeInformation HOB which broke logic in subsequent consumers. Removed a spurious allocation that was unused and never freed. Cleaned up coding style to match ECC.

DxeCheckMemoryMap - 
fixed an incorrect index that lead to wrong information display

makubacki commented:
@joschock, as we discussed outside the PR, it would be nice if the changes could be broken into two separate commits per file. Then we can rebase & ff the changes in. Leaving the uncrustify changes in with the functional changes should be fine.

joschock committed:
MinPlatform/TestPointCheckLib: Fix incorrect index in MemoryMap test point.

Resolve an issue where an incorrect index was used leading to wrong information display.

joschock committed:
MinPlatform/TestPointCheckLib: Fix incorrect modification of MemoryTypeInformation HOB and memory leak.

Fixed an issue where MemoryTypeInfo check was incorectly modifying the MemoryTypeInformation HOB which broke logic in subsequent consumers.
Also removed a spurious allocation that was unused and never freed.

joschock commented:
> @joschock, as we discussed outside the PR, it would be nice if the changes could be broken into two separate commits per file. Then we can rebase & ff the changes in. Leaving the uncrustify changes in with the functional changes should be fine.

Split into two commits per our discussion.
